lib/ecto/adapters/clickhouse/migration.ex

```diff
@@ -305,6 +328,29 @@ defmodule Ecto.Adapters.ClickHouse.Migration do
   defp index_expr(literal) when is_binary(literal), do: literal
   defp index_expr(literal), do: @conn.quote_name(literal)
 
+  defp constraint_expr(%Constraint{check: check, validate: true, comment: nil} = constraint)
+       when is_binary(check) do
+    [
+      @conn.quote_name(constraint.name),
+      " CHECK (",
+      check,
+      ")"
+    ]
+  end
+
+  defp constraint_expr(%Constraint{check: check, validate: true, comment: comment})
+       when is_binary(check) and is_binary(comment) do
+    raise "Clickhouse adapter does not support comments on check constraints"
```

Comment:
I'd probably put all this checks into a single function clause like this

```elixir
defp constraint_expr(%Constraint{} = constraint) do
  if constraint.comment do
    raise "Clickhouse adapter does not support comments on check constraints"
  end
  
  unless constraint.validate do
    raise ...
  end
  
  # etc. checks for other unsupported fields / options
  
  [
       @conn.quote_name(constraint.name),
       " CHECK (",
       check,
       ")"
     ]
end
```

Reply:
In that case maybe it's more appropriate to move them inside execute_ddl/1